src/features/groups/Edit/index.tsx

```diff
@@ -13,65 +13,76 @@ import Members from '../Members';
 import { ROUTES } from 'constants/routes';
 
 import { ReactComponent as BackIconSVG } from 'assets/images/back-icon.svg';
-import { ReactComponent as ArrowRightSVG } from 'assets/images/arrow-right.svg';
+import { SliderItemsWrapper, SliderNavItem } from './index.styled';
 
-import { GroupEditContentWrapper } from './index.styled';
-import Loader from 'components/Loader';
-import { ButtonBase, TextUnderline } from 'components';
-
-enum GroupEditMode {
-  GeneralInfo,
-  Members,
-}
+import Slider from "react-slick";
 
 const Edit = () => {
   const { t } = useTranslation('groups');
   const navigate = useNavigate();
   const { groupId } = useParams();
+  const sliderItems = groupId ? ["mainInfo", "members", "statistics"] : ["mainInfo"];
 
-  const [getGroup, { data: group, isLoading }] = useLazyGetTransactionGroupByIdQuery();
-
-  const [mode, setMode] = useState<GroupEditMode>(GroupEditMode.GeneralInfo);
+  const [getGroup, { isLoading, data: group }] = useLazyGetTransactionGroupByIdQuery();
+  const [activeSliderItem, setActiveSliderItem] = useState<string>(sliderItems[0]);
 
   const handleBackAction = useCallback(() => {
     navigate(ROUTES.GROUPS.ROOT, { replace: true });
   }, [navigate]);
 
-  const onChangeModeHandler = () => {
-    setMode(mode => mode === GroupEditMode.GeneralInfo ? GroupEditMode.Members : GroupEditMode.GeneralInfo);
-  }
 
   useEffect(() => {
     if (groupId) {
       getGroup(groupId);
     }
   }, [groupId, getGroup]);
 
+  let sliderRef = useRef<Slider>(null);
+
+  var settings = {
+    dots: false,
+    infinite: false,
+    speed: 500,
+    slidesToShow: 1,
+    slidesToScroll: 1,
+    adaptiveHeight: false,
+    className: "w-100",
+    beforeChange: (_: number, next: number) => setActiveSliderItem(sliderItems[next])
+  };
+
+  const sliderPagesList = (sliderItems: Array<string>) => {
+    return <SliderItemsWrapper>
+      {sliderItems.map((item, idx) => {
+        return (
+          <SliderNavItem 
+            isActive={activeSliderItem === item} 
+            onClick={() => { setActiveSliderItem(item); sliderRef.current?.slickGoTo(idx); }}
+          >
+            {t(item + 'SliderItem')}
+          </SliderNavItem>
+          )
+      })}
+    </SliderItemsWrapper>
+
+  }
```

Comment:
Would it be possible to move sliderPagesList function outside of the component to avoid its retendering each time?

Reply:
It’d be better to do that, yes this way, yes.